transcendence-app/src/shared/local-file/local-file.interceptor.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable, mixin, NestInterceptor, Type } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { MulterOptions } from '@nestjs/platform-express/multer/interfaces/multer-options.interface';
+import { diskStorage } from 'multer';
+import { join } from 'path';
+import { LocalFileConfig } from './local-file.config.interface';
+
+interface LocalFileInterceptorOptions {
+  fieldName: string;
+  path?: string;
+  fileFilter?: MulterOptions['fileFilter'];
+  limits?: MulterOptions['limits'];
+}
+
+function LocalFileInterceptor(
+  options: LocalFileInterceptorOptions,
+): Type<NestInterceptor> {
```

Comment:
TLDR: It's extending the `FileInterceptor` to use a default directory for storing the files and to optionally add a subpath, filter the file by MIME type or limit by file size.

More info about how this works -> https://wanago.io/2021/11/08/api-nestjs-uploading-files-to-server/ under 'Extending the FileInterceptor'